src/hotspot/share/code/nmethod.cpp

```diff
@@ -1525,96 +1333,69 @@ bool nmethod::make_not_entrant_or_zombie(int state) {
   }
 #endif
 
-  // When the nmethod becomes zombie it is no longer alive so the
-  // dependencies must be flushed.  nmethods in the not_entrant
-  // state will be flushed later when the transition to zombie
-  // happens or they get unloaded.
-  if (state == zombie) {
-    {
-      // Flushing dependencies must be done before any possible
-      // safepoint can sneak in, otherwise the oops used by the
-      // dependency logic could have become stale.
-      MutexLocker mu(CodeCache_lock, Mutex::_no_safepoint_check_flag);
-      if (nmethod_needs_unregister) {
-        Universe::heap()->unregister_nmethod(this);
-      }
-      flush_dependencies(/*delete_immediately*/true);
```

Comment:
`flush_dependencies` is now only ever called with `false`

Reply:
Good catch. I removed it, and some remove_dependent methods that were only used in the true version that is no longer called.